pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -606,14 +681,15 @@ cdef class {{name}}HashTable(HashTable):
                                  ignore_na=True, return_inverse=True)
         return labels
 
+    {{if dtype == 'int64'}}
```

Comment:
why adding this only for int64?

Reply:
This function is used just in int64-version:

https://github.com/pandas-dev/pandas/blob/aad85ad1e39568f87cdae38fece41cd2370234a7/pandas/core/sorting.py#L598-L603

The logic in this function: https://github.com/pandas-dev/pandas/blob/1fc5efd76ea1f85979fa2364a292d85482e338aa/pandas/_libs/hashtable_class_helper.pxi.in#L628-L633

IIUC, negative numbers are special and should be skipped, but this is only a convention for the meaning of values `group_index`. For other types there is no such convention. Instead of inventing something, I've decided to delete unused functions/versions.